### PR TITLE
Cancel button behavior for showdatepicker/showtimepicker

### PR DESCRIPTION
**Pull request by warent**
Issue #6483

### CODE REVIEW

File: packages/flutter/lib/src/material/date_picker.dart [hunk 1]
```diff
@@ -691,5 +690,5 @@ Future<DateTime> showDatePicker({
       lastDate: lastDate
     )
   );
-  return picked ?? initialDate;
+  return picked ?? null;
```
Comment (abarth) [inline packages/flutter/lib/src/material/date_picker.dart]: There's no need to `?? null`.  You can just return `picked`.  The `??` operator returns the right-hand side when the left-hand side is null.  If the left-hand side is null, then we can just return that null.

Comment (abarth) [inline packages/flutter/lib/src/material/date_picker.dart]: I'd also skip the local variable and just `return await showDialog...`

File: packages/flutter/lib/src/material/time_picker.dart [hunk 2]
```diff
@@ -732,8 +731,10 @@ Future<TimeOfDay> showTimePicker({
   TimeOfDay initialTime
 }) async {
   assert(initialTime != null);
-  return await showDialog(
+  TimeOfDay picked = await showDialog(
     context: context,
     child: new _TimePickerDialog(initialTime: initialTime)
-  ) ?? initialTime;
```
Comment (abarth) [inline packages/flutter/lib/src/material/time_picker.dart]: Same comments here.  We can return the result of the `await` directly without the `??` or the local variable.

File: packages/flutter/lib/src/material/date_picker.dart (hunk 1 above)
Comment (warent) [inline packages/flutter/lib/src/material/date_picker.dart]: Fair enough. I thought the null return would appear more transparent and obviously intentional when written that way, but I can update :)

**Comment by abarth** [comment]
LGTM

Please email flutter-dev after this patch lands to explain the API change.

**Pull request merged**